core/src/main/java/com/linecorp/armeria/internal/ArmeriaHttpUtil.java

```diff
@@ -489,8 +515,18 @@ public static long parseCacheControlSeconds(@Nullable String value) {
     /**
      * Converts the specified Netty HTTP/2 into Armeria HTTP/2 headers.
      */
-    public static HttpHeaders toArmeria(Http2Headers headers, boolean endOfStream) {
-        final HttpHeaders converted = new DefaultHttpHeaders(false, headers.size(), endOfStream);
+    public static HttpHeaders toArmeria(Http2Headers headers, boolean request, boolean endOfStream) {
+        final HttpHeadersBuilder converted;
+        if (request) {
+            converted = headers.contains(HttpHeaderNames.METHOD) ? new RequestHeadersBuilder(headers.size())
```

Comment:
nit: `requireNonNull(headers, "headers");`

Reply:
It's an internal use only API, so I'd like to skip null check here.